cli/command/service_test.go

```diff
@@ -0,0 +1,322 @@
+package command
+
+import (
+	"net/url"
+	"testing"
+
+	"github.com/quintilesims/layer0/common/models"
+	"github.com/urfave/cli"
+)
+
+func TestCreateService(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+	serviceCommand := NewServiceCommand(base.Command())
+
+	req := models.CreateServiceRequest{
+		DeployID:       "dpl_id",
+		EnvironmentID:  "env_id",
+		LoadBalancerID: "lb_id",
+		ServiceName:    "svc_name",
+	}
+
+	base.Resolver.EXPECT().
+		Resolve("deploy", "dpl_name").
+		Return([]string{"dpl_id"}, nil)
+
+	base.Resolver.EXPECT().
+		Resolve("environment", "env_name").
+		Return([]string{"env_id"}, nil)
+
+	base.Resolver.EXPECT().
+		Resolve("load_balancer", "lb_name").
+		Return([]string{"lb_id"}, nil)
+
+	base.Client.EXPECT().
+		CreateService(req).
+		Return("job_id", nil)
+
+	base.Client.EXPECT().
+		ReadJob("job_id").
+		Return(&models.Job{
+			Status: "Completed",
+			Result: "svc_id",
+		}, nil)
+
+	base.Client.EXPECT().
+		ReadService("svc_id").
+		Return(&models.Service{}, nil)
+
+	flags := map[string]interface{}{
+		"loadbalancer": "lb_name",
+	}
+
+	c := getCLIContext(t, []string{"env_name", "svc_name", "dpl_name"}, flags)
```

Comment:
With the above comment on the variables in mind, this would now read:

`c := getCLIContext(t, args{"env_name", "svc_name", "dpl_name"}, flags)`

Or, you could just inline flags as you have only one pair:
`c := getCLIContext(t, args{"env_name", "svc_name", "dpl_name"}, flags{"loadbalancer": "lb_name"})`

Similar substitutions can be made on other `getCLIContext()` calls

Reply:
@jparsons04 I may not have args and flags objects just yet, but assuming that I do: have you (& others) been preferring to declare both args and flags before the call to getCLIContext, or inline?

In other words:
```
args := args{"env_name", "svc_name", "dpl_name"}
flags := flags{"loadbalancer": "lb_name"}
c := getCLIContext(t, args, flags)
```
or
```
c := getCLIContext(t, args{"env_name", "svc_name", "dpl_name"}, flags{"loadbalancer": "lb_name"})
```

Reply:
I might need more clarification on it, but I think I've been assigning things to variables if they are multi-line or would make the call to the function too long. I could conceivably make a case for assigning Args{} and Flags{} as variables before calling `getCLIContext`

Reply:
Tests have been updated to define `Args{}` and `Flags{}` before passing them into the command functions. ✅